Source/Apple/arcana/tracing/trace_region.h

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright (C) Microsoft Corporation. All rights reserved.
+//
+
+#pragma once
+
+#include <os/signpost.h>
+
+namespace arcana
+{
+    class trace_region final
+    {
+    public:
+        trace_region() = delete;
+        trace_region& operator=(const trace_region&) = delete;
+        trace_region(const trace_region&) = delete;
+        trace_region& operator=(trace_region&&) = delete;
```

Comment:
nit: This is a weird order. Maybe constructors first, then assignments? Or at least constructor/assignment for copy, then for move?